odoo/addons/base/models/ir_binary.py

```diff
@@ -0,0 +1,164 @@
+        elif record._name == 'ir.attachment':
+            stream = Stream.from_attachment(record)
+        elif not field_def.attachment or field_def.compute or field_def.related:
+            stream = Stream.from_binary_field(record, field)
+        else:
+            field_attachment = self.env['ir.attachment'].sudo().search(
+                domain=[('res_model', '=', record._name),
+                        ('res_id', '=', record.id),
+                        ('res_field', '=', field)],
+                limit=1)
+            if not field_attachment:
+                raise MissingError("The related attachment does not exist.")
+            stream = Stream.from_attachment(field_attachment)
+
+        if stream.type in ('data', 'path'):
+            if mimetype:
+                stream.mimetype = mimetype
+            elif not stream.mimetype:
+                if stream.type == 'data':
+                    head = stream.data[:1024]
+                else:
+                    with open(stream.path, 'rb') as file:
+                        head = file.read(1024)
+                stream.mimetype = guess_mimetype(head, default=default_mimetype)
+
+            if filename:
+                stream.download_name = filename
+            elif filename_field in record:
+                stream.download_name = record[filename_field]
+            if not stream.download_name:
+                stream.download_name = f'{record._table}-{record.res_id}-{field}'
+
+            if not os.path.splitext(stream.download_name)[1]:  # missing file extension
+                stream.download_name += guess_extension(stream.mimetype) or ''
+
+        return stream
+
+    def _get_image_stream_from(
+        self, record, field='raw', filename=None, filename_field='name',
+        mimetype=None, default_mimetype='image/png', placeholder=None,
+        width=0, height=0, crop=False, quality=0,
+    ):
+        try:
+            stream = self._get_stream_from(record, field, filename, filename_field, mimetype, default_mimetype)
+        except werkzeug.exceptions.NotFound:
+            if request.params.get('download'):
+                raise
+            if not placeholder:
+                placeholder = record._get_placeholder_filename(field)
+            stream = self._get_placeholder_stream(placeholder)
+
+        if stream.type == 'url':
+            return stream  # Rezising an external URL is not supported
+
+        if (width, height) == (0, 0):
+            width, height = image_guess_size_from_field_name(field)
+        stream.etag += f'-{width}x{height}-crop={crop}-quality={quality}'
+
+        if isinstance(stream.last_modified, (int, float)):
+            stream.last_modified = datetime.utcfromtimestamp(stream.last_modified)
+        modified = werkzeug.http.is_resource_modified(
+            request.httprequest.environ,
+            etag=stream.etag,
+            last_modified=stream.last_modified
+        )
+
+        if modified and (width or height or crop):
```

Comment:
A comment should be added. For example:
If the source image has not been modified then there is no need to read it to resize it because the resize will also be identical.